src/inmanta/protocol/methods_v2.py

```diff
@@ -1357,3 +1357,73 @@ def set_password(username: str, password: str) -> None:
     :raises NotFound: Raised when the user does not exist
     :raises BadRequest: Raised when server authentication is not enabled
     """
+
+
+@typedmethod(
+    path="/unmanaged/<unmanaged_resource_id>",
+    validate_sid=False,
+    operation="POST",
+    agent_server=True,
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.agent],
+    api_version=2,
+)
+def unmanaged_resource_create(env: uuid.UUID, unmanaged_resource_id: str, values: Dict[str, str]) -> None:
+    """
+    create a discovered resource in the DB
+    :param env: The id of the environment this resource belongs to
+    :param unmanaged_resource_id: The id of the unmanaged resource
+    :param values: The values associated with the unmanaged_resource
+    """
+
+
+@typedmethod(
+    path="/unmanaged/",
+    validate_sid=False,
+    operation="POST",
+    agent_server=True,
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.agent],
+    api_version=2,
+)
+def unmanaged_resource_create_batch(env: uuid.UUID, unmanaged_resources: List[UnmanagedResource]) -> None:
+    """
+    create multiple discovered resource in the DB
+    :param env: The id of the environment this resource belongs to
+    :param unmanaged_resources: List of UnmanagedResource containing the unmanaged_resource_id and values for each resource
+    """
+
+
+@typedmethod(
+    path="/unmanaged/<unmanaged_resource_id>",
+    operation="GET",
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.api],
+    api_version=2,
+)
+def unmanaged_resources_get(tid: uuid.UUID, unmanaged_resource_id: str) -> model.UnmanagedResource:
+    """
+    Get a single discovered resource
+    :param tid: the id of the environment in which to get the unmanaged resource.
+    :param unmanaged_resource_id: The id of the unmanaged resource
+    """
+
+
+@typedmethod(
+    path="/unmanaged/",
+    operation="GET",
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.api],
+    api_version=2,
+)
+def unmanaged_resources_get_batch(
+    tid: uuid.UUID, start_resource_id: Optional[str] = None, end_resource_id: Optional[str] = None, limit: Optional[int] = None
+) -> List[model.UnmanagedResource]:
+    """
+    Get the unmanaged resources with paging
+    :param tid: the id of the environment in which to get the unmanaged resources.
+    :param start_resource_id: the unmanaged_resource_id from which to start the paging
+    :param end_resource_id: the unmanaged_resource_id where to finish the paging
+    :param limit: the maximal number of resources to return
```

Comment:
This paging API look different from all the other paging APIs we already have. It for example doesn't have a field to indicate the order in which we want to page the output. Is that done on purpose?

Reply:
Indeed. It seems our way of paging is not as standardized as I thought it way. That's a shame. You can ignore this comments.

Reply:
well I should use DataView (see wouters comment) so we do have a standard